apps/admin/frontend/src/components/smartcard_modal/smartcard_modal.test.tsx

```diff
@@ -61,11 +51,11 @@ afterEach(() => {
 });
 
 test('Smartcard modal displays card details', async () => {
-  renderRootElement(<App hardware={mockHardware} />, {
-    apiClient: mockApiClient,
-    backend: mockBackend,
-  });
-  await authenticateAsSystemAdministrator(mockApiClient);
+  const { apiMock, renderApp } = buildApp(mockApiClient);
```

Comment:
Some of them could, but I think the special thing about the `smartcard_modal` is that it can pop up anywhere in the app and is nested at a high level. I don't intend to touch them.

Reply:
That's fair, though the "popping up anywhere" could be tested once at the app level, and then all the detailed tests could be just for the modal. But fine to kick that down the road for later.